rebasebot/bot.py

```diff
@@ -410,24 +419,34 @@ def _init_working_dir(
 
     logging.info("Fetching %s from dest", dest.branch)
     gitwd.remotes.dest.fetch(dest.branch)
+
     logging.info("Fetching %s from source", source.branch)
     gitwd.remotes.source.fetch(source.branch)
 
+    logging.info("Fetching all tags from source")
+    gitwd.remotes.source.fetch(refspec='refs/tags/*:refs/tags/*')
+
+    if is_ref_a_tag(gitwd, source.branch):
```

Comment:
I wondering if we should deal with possible names collision here.

If some branch and tag have the same names, tag would take precedence now as far as I understand.

Reply:
Yes. Which I think it's desirable.
Release branches (which both upstream Cloud Providers and CAPI Providers use) follow the `release-X.Y` format, while tags follow `vX.Y.Z(-.*)?` format. I think those normally should not collide, and if they do, I would favour tags over branches as in general i think we should try to rebase on top of specific versions rather than a more vague commit stream in a release branch.

Reply:
WDTY?

Reply:
tricky question, i think giving the tag precedence is ok. theoretically, if projects are creating a `release-X.Y` branch and then a tag `release-X.Y`, i would expect the branch to receive updates and then for followup tags to be created, eg `release-X.Y.Z`. but even if this were to happen, i would still want to prefer the tag for `release-X.Y` over the branch (and i realized this would be a naming error most likely as the tag should have been `release-X.Y.0` or something).

Reply:
Agree